src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AssigningSymbolAndItsMemberInSameStatement.cs

```diff
@@ -0,0 +1,97 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftQualityGuidelinesAnalyzersResources.AssigningSymbolAndItsMemberInSameStatementDescription), MicrosoftQualityGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftQualityGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            s_localizableDescription);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterOperationAction(AnalyzeAssignment, OperationKind.SimpleAssignment);
+        }
+
+        private void AnalyzeAssignment(OperationAnalysisContext context)
+        {
+            var assignmentOperation = (ISimpleAssignmentOperation)context.Operation;
+
+            // Check if there are more then one assignment in a statement
+            if (!(assignmentOperation.Target is IMemberReferenceOperation operationTarget))
+            {
+                return;
+            }
+
+            // This analyzer makes sense only for reference type objects
+            if (operationTarget.Instance?.Type.IsValueType == true)
+            {
+                return;
+            }
+
+            // Search for object equal to operationTarget.Instance further in assignment chain
+            bool isViolationFound = false;
+            if (operationTarget.Instance is ILocalReferenceOperation localInstance)
+            {
+                isViolationFound = AnalyzeAssignmentToMember(assignmentOperation, localInstance, (a, b) => a.Local == b.Local);
+            }
+            else if (operationTarget.Instance is IMemberReferenceOperation memberInstance)
+            {
+                isViolationFound = AnalyzeAssignmentToMember(assignmentOperation, memberInstance, (a, b) => a.Member == b.Member && a.Instance?.Syntax.ToString() == b.Instance?.Syntax.ToString());
+            }
+            else if (operationTarget.Instance is IParameterReferenceOperation parameterInstance)
+            {
+                isViolationFound = AnalyzeAssignmentToMember(assignmentOperation, parameterInstance, (a, b) => a.Parameter == b.Parameter);
+            }
+            else
+            {
+                return;
+            }
+
+            if (isViolationFound)
+            {
+                var diagnostic = Diagnostic.Create(Rule, operationTarget.Syntax.GetLocation(), operationTarget.Instance.Syntax.ToString());
```

Comment:
The new message is `Symbol '{0}' and its member '{1}' are both assigned in the same statement. You are at risk of assigning the member of an unintended object.`, which has 2 arguments, but the diagnostic created here passes a single argument. Either remove '{1}' from the message or update the diagnostic and all unit tests to have the member name.

Reply:
Right. Didn't notice it. Hm. But if expression is something like `a.b.c = a.b = d`, then what is the member? If it's, just `b`, then this message could cause some confusion. I will change it to `Symbol 'a' and its member 'a.b' are both assigned in the same statement. You are at risk of assigning the member of an unintended object.`

Reply:
Is it possible to generate `Symbol 'a' and its member 'b' are both assigned in the same statement. You are at risk of assigning the member of an unintended object.`? Otherwise, I would be fine with changing the message to `Symbol '{0}' and its member are both assigned in the same statement. You are at risk of assigning the member of an unintended object.`

Reply:
Actually, even `Symbol 'a' and its member 'a.b' are both assigned in the same statement. You are at risk of assigning the member of an unintended object` looks fine. I would let you make a call.

Reply:
In the end, I think your way is better. Implemented it as `Symbol 'a' and its member 'b' are both assigned in the same statement. You are at risk of assigning the member of an unintended object.`